### DISCUSSION TITLE
Why can't I use the U8G2 library with the Constructor HW?

### DISCUSSION THREAD

**Discussion by Pepe844** [olikraus/u8g2#2096] [General]
I have a project based on a Heltec board ESP32-S3 with onboard display (dollatek 2.42-inch 128x64-oled - ssd1309 driven display). 
On the one hand the onboard display works with the U8G2 library (constructor HW) fine. The external display, on the other hand, can only be addressed with the Constructor SW, unfortunately the application is too slow. Is there a way to address the external display with the constructor HW as well?

**Comment by olikraus** [comment]
You might try the HW constructor remapping feature: 

..._HW_I2C(U8X8_PIN_NONE, clock, data)

This might be supported by the ESP32 hardware (not 100% sure).
